### ISSUE TITLE
Qt ItemDelegate should override initStyleOption

### ISSUE THREAD

**Issue by brownnrl** [hsoft/moneyguru#411]
In the import autofill ticket I've been experimenting with setting the background color of a changed item.  This is with respect to ``qt.support.item_delegate.ItemDelegate``.

I had a lot of trouble with doing this, and I couldn't understand why setting the ``backgroundBrush`` or certain other attributes of the ``option`` wasn't working.  I went so far as to dig into the Qt code to find the [base ``paint`` method](https://qt.gitorious.org/qt/qt/source/25d972e12eda9dadf212d24af8d8f524572bdbfa:src/gui/itemviews/qstyleditemdelegate.cpp#L419).

After review, it was clear that the base ``QStyledItemDelegate.paint`` method calls ``initStyleOption`` before painting the control, which resets certain options set by the ``ItemDelegate`` subclass with their defaults just prior to painting.  The ``ItemDelegate`` subclass should override the ``initStyleOption`` method to ensure that the options to be set are correct.  See also this [SO issue](http://stackoverflow.com/questions/10219739/set-color-to-a-qtableview-row).

This isn't a bug, but it is a design defect that could bite later if not corrected in master (or at least develop?).  I will submit a PR later, but I wanted to document it as an issue in case I don't get around to it for a bit.

**Comment by brownnrl** [comment]
You can see a first attempt from the above referenced commit on the #26 branch (still experimental).

**Comment by hsoft** [comment]
Live and learn! Nice to know, that `initStyleOption` thing.

**Comment by hsoft** [comment]
I wanted to cherry-pick your commit @brownnrl , but it doesn't apply cleanly because it references stuff from your experimental branch (`pre_paint` and stuff...). But since I'd like to fix this before the v2.9 release, I reworked it to apply cleanly on `develop`. I also removed `_prepare_paint_options()` which don't make any sense anymore because it serves the same purpose as `initStyleOption()`.

But thanks for the hint!
